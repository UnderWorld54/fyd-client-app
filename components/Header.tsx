import { StyleSheet, Text, View } from 'react-native';
import BackButton from './BackButton';

type HeaderProps = {
  title: string;
};

export default function Header({ title }: HeaderProps) {
  return (
    <View style={styles.header}>
      <BackButton />
      <Text style={styles.title} numberOfLines={1} ellipsizeMode="tail">{title}</Text>
      <View style={{ width: 40 }} />
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 60,
    marginTop: 32,
    marginBottom: 24,
    paddingHorizontal: 8,
  },
  title: {
    flex: 1,
    textAlign: 'center',
    fontSize: 24,
    fontWeight: 'bold',
    fontFamily: 'MontserratBold',
  },
}); 