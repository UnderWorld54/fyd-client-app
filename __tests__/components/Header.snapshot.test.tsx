import Header from '@/components/Header';
import { render } from '@testing-library/react-native';
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

describe('Header Component Snapshots', () => {
  it('matches snapshot with title only', () => {
    const tree = render(<Header title="Test Header" />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with right component', () => {
    const rightComponent = (
      <TouchableOpacity>
        <Text>Action</Text>
      </TouchableOpacity>
    );
    
    const tree = render(
      <Header title="Header with Action" rightComponent={rightComponent} />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with very long title', () => {
    const longTitle = 'This is a very long title that should be truncated with ellipsis at the end';
    const tree = render(<Header title={longTitle} />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});