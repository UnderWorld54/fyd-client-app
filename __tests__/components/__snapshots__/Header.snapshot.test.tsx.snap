// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshots matches snapshot with right component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "marginBottom": 24,
      "marginTop": 32,
      "paddingHorizontal": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#000"
      name="chevron-back"
      size={24}
    />
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "flex": 1,
        "fontFamily": "MontserratBold",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Header with Action
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Action
      </Text>
    </View>
  </View>
</View>
`;

exports[`Header Component Snapshots matches snapshot with title only 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "marginBottom": 24,
      "marginTop": 32,
      "paddingHorizontal": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#000"
      name="chevron-back"
      size={24}
    />
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "flex": 1,
        "fontFamily": "MontserratBold",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Test Header
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 40,
      }
    }
  />
</View>
`;

exports[`Header Component Snapshots matches snapshot with very long title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "marginBottom": 24,
      "marginTop": 32,
      "paddingHorizontal": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#000"
      name="chevron-back"
      size={24}
    />
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "flex": 1,
        "fontFamily": "MontserratBold",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    This is a very long title that should be truncated with ellipsis at the end
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 40,
      }
    }
  />
</View>
`;
